### PR TITLE
FIX #72, config space now mandatory for scenario

### LINKED ISSUE #72: Make configuration space a mandatory object
Currently, it is possible to create a scenario object without passing a configuration space object to the scenario. SMAC then fails later on. Will create a PR soonish.

### PR DESCRIPTION
**Pull request by mfeurer**
This PR makes sure that either the argument `cs` or `paramfile` is passed to the scenario object.

### CODE REVIEW

File: smac/scenario/scenario.py [hunk 1]
```diff
@@ -64,13 +66,48 @@ def __init__(self, scenario, cmd_args=None):
             raise ValueError('Could not parse the following arguments: %s' %
                              str(list(scenario.keys())))
 
+        for group, potential_members in self._groups.items():
```
Comment (KEggensperger) [inline smac/scenario/scenario.py]: If an argument is also set as required this check is not influenced and the code might crash before?
This should be clarified in the docstring: if an argument is part of the required group one should not mark it as required.

File: smac/scenario/scenario.py [hunk 2]
```diff
@@ -64,13 +66,48 @@ def __init__(self, scenario, cmd_args=None):
...
+            If given, the callback will be called when the argument is
+            parsed. Useful for custom casting/typechecking.
+        default : object, optional
+            Default value if the argument is not given. Default to ``None``.
+        dest : str
+            Assign the argument to scenario object by this name.
+        required : bool
+            If True, the scenario will raise an error if the argument is not
+            given.
+        required_group : str
```
Comment (KEggensperger) [inline smac/scenario/scenario.py]: The name is misleading. What about mutually_exclusive_group?

File: smac/scenario/scenario.py (hunk 1 above)
Comment (mfeurer) [inline smac/scenario/scenario.py]: I added a check for that and adapted the docstring.

File: smac/scenario/scenario.py (hunk 2 above)
Comment (mfeurer) [inline smac/scenario/scenario.py]: I changed the name of the argument.

**Pull request merged**